src/precice/Participant.hpp

```diff
@@ -22,27 +22,180 @@ struct WhiteboxAccessor;
 
 namespace precice {
 
-/// convenience type for compatibility
+/** @brief forwards-compatible typedef for C++17 \ref std::string_view.
+ *
+ * The \ref string_view is a read-only view into sequence of characters.
+ *
+ * A good mental model for this type is a struct containing a pointer and a size like the following:
+ *
+ * @code{cpp}
+ * struct string_view {
+ *   const char* first;
+ *   size_t size;
+ * };
+ * @endcode
+ *
+ * It can be constructed from:
+ * - a pointer to a NULL-terminated C-string
+ * - a pointer and a size
+ * - a pointer to the first and one past the last element
+ * - \ref std::string can directly convert itself to \ref std::string_view.
+ *
+ * In practise, using a function such as \ref Participant::getMeshDimensions() that expects a string_view looks like this:
+ *
+ * You can directly pass the string literal:
+ *
+ * @code{.cpp}
+ * getMeshDimensions("MyMesh");
+ * @endcode
+ *
+ * Or you can pass something string-like:
+ *
+ * @code{.cpp}
+ * // one of
+ * const char*      meshName = "MyMesh";
+ * std::string      meshName = "MyMesh";
+ * std::string_view meshName = "MyMesh";
+ * // followed by
+ * getMeshDimensions(meshName);
+ * @endcode
+ *
+ * For more complex settings, you can pass pointer + size:
```

Comment:
What are those scenarios? This looks more like: "you can use the following constructor", but I could imagine that it confuses more than it helps.

Reply:
I was thinking of such C-style functions:

```cpp
char meshName[MAX_MESH_NAME];
int length = getMeshNameFromObscurePlace(meshName);
getMeshDimensions(precice::string_view{meshName, length});
```

This is how MPI provides access to opened ports work for example.

Reply:
>  then we can also remove it.

That's no strong opinion from my side. For C codes, we anyway have a different interface.

Reply:
Sure, I can remove it. Less is sometimes more.